Alpaca.Markets/SockClient.cs

```diff
@@ -61,13 +59,14 @@ public sealed partial class SockClient : IDisposable
             {
                 Scheme = alpacaRestApi.Scheme == "http" ? "ws" : "wss"
             };
-            uriBuilder.Path += "/stream";
-
-            _webSocket = webSocketFactory.CreateWebSocket(uriBuilder.Uri);
 
-            _webSocket.Opened += handleOpened;
-            _webSocket.Closed += handleClosed;
+            if (uriBuilder.Path.Substring(uriBuilder.Path.Length - 1, 1) != "/")
+            {
+                uriBuilder.Path += "/";
+            }
+            uriBuilder.Path += "stream";
 
+            setupWebSocket(uriBuilder, webSocketFactory);
             _webSocket.DataReceived += handleDataReceived;
```

Comment:
You never unsubscribe these events. In this particular case it's not a problem but in general - it's not a good idea.

Reply:
They're removed in the base factory class' dispose method, right? I'm not sure where else you'd want that done.

Reply:
No. You have to unsubscribe events explicitly in Dispose method. Currently Dispose pattern implemented incorrectly. I'll fix it in separate commit as well as other `SockClientBase` issues.